cmd/collector/collector_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+)
+
+var testFlowRecords = []string{"flow1", "flow2", "flow3"}
+
+func TestFlowRecordHandler(t *testing.T) {
+	flowRecords = testFlowRecords
+	defer func() {
+		flowRecords = nil
+	}()
+
+	testCases := []struct {
+		name           string
+		countParam     string
+		formatParam    string
+		expectedStatus int
+		expectedFlows  []string
+	}{
+		{
+			name:          "default",
+			expectedFlows: []string{"flow1", "flow2", "flow3"},
+		},
+		{
+			name:          "last flow",
+			countParam:    "1",
+			expectedFlows: []string{"flow3"},
+		},
+		{
+			name:          "text format",
+			formatParam:   "text",
+			expectedFlows: []string{"flow1", "flow2", "flow3"},
+		},
+		{
+			name:          "json format",
+			formatParam:   "json",
+			expectedFlows: []string{"flow1", "flow2", "flow3"},
+		},
+		{
+			name:          "large count",
+			countParam:    "100",
+			expectedFlows: []string{"flow1", "flow2", "flow3"},
+		},
+		{
+			name:           "invalid count",
+			countParam:     "-1",
+			expectedStatus: http.StatusBadRequest,
+		},
+		{
+			name:           "invalid format",
+			formatParam:    "foobar",
+			expectedStatus: http.StatusBadRequest,
+		},
+		{
+			name:          "both params set",
+			countParam:    "2",
+			formatParam:   "text",
+			expectedFlows: []string{"flow1", "flow2"},
+		},
+	}
+
+	for _, tc := range testCases {
+		expectedStatus := tc.expectedStatus
+		if expectedStatus == 0 {
+			expectedStatus = http.StatusOK
+		}
+		format := tc.formatParam
+		if format == "" {
+			format = "json"
+		}
+		rr := httptest.NewRecorder()
+		u := url.URL{
+			Path: "/records",
+		}
+		q := u.Query()
+		if tc.countParam != "" {
+			q.Set("count", tc.countParam)
+		}
+		if tc.formatParam != "" {
+			q.Set("format", tc.formatParam)
+		}
+		u.RawQuery = q.Encode()
+		req, err := http.NewRequest("GET", u.String(), nil)
+		require.NoError(t, err)
+		flowRecordHandler(rr, req)
+		resp := rr.Result()
+		defer resp.Body.Close()
+		require.Equal(t, expectedStatus, resp.StatusCode)
+		body, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+
+		if expectedStatus != http.StatusOK {
+			return
```

Comment:
I think this return will ignore the last 3 tests.

Reply:
yes, that's correct, good catch
I forgot to call `t.Run` to define the subtests